### PR TITLE
Added support for visualiser feature

### PR DESCRIPTION
**Pull request by vikiCoder**
- Created Visualiser module that provides the functionality to process templates using handlebars.
- Used this Visualiser module to process templates passed as the result of script execution from the sandbox.
- Passed the processed template along with given options and processing errors in `result.return.visualiser` of `script` callback

Todo:
- [x] Bump postman-sandbox version with https://github.com/postmanlabs/postman-sandbox/pull/464 merged.

### CODE REVIEW

File: lib/visualiser/index.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        try {
+            jsonResponse = response && response.json();
+        }
+        catch (err) {
+            // ignore response if it is not a valid json response
+            jsonResponse = undefined;
+        }
+
+        // merge all data according to their priority into single object
+        templateData = _.merge({},
```
Comment (codenirvana) [inline lib/visualiser/index.js]: maybe this is not required

File: lib/visualiser/index.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            (response = new sdk.Response(response));
+
+        var handlebarsOptions = options && options.handlebarsOptions,
+            compiledTemplate = Handlebars.compile(template, handlebarsOptions),
+            jsonResponse,
+            templateData,
+            processedTemplate;
+
+        try {
+            jsonResponse = response && response.json();
```
Comment (codenirvana) [inline lib/visualiser/index.js]: not required, let user handle this in `options.data`

File: lib/visualiser/index.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+        try {
+            // hydrate the template with provided data
+            processedTemplate = compiledTemplate(templateData);
+        }
+        catch (err) {
+            return callback(err);
+        }
+
+        return callback(null, processedTemplate, templateData);
```
Comment (codenirvana) [inline lib/visualiser/index.js]: Why `templateData`?

File: lib/visualiser/index.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        if (!(template && typeof template === 'string')) {
+            return callback(new Error(`Invalid template. Template must be of type string, found ${typeof template}`));
+        }
+
+        if (!callback && typeof options === 'function') {
+            callback = options;
+            options = {};
+        }
+
+        typeof callback !== 'function' && (callback = _.noop);
```
Comment (codenirvana) [inline lib/visualiser/index.js]: these safety checks are not required for private method.

File: lib/visualiser/index.js (hunk 3 above)
Comment (vikiCoder) [reply lib/visualiser/index.js]: It is needed to provide access to data inside iframe in the app

File: lib/runner/extensions/event.command.js [hunk 5]
```diff
@@ -410,6 +413,25 @@ module.exports = {
                     // upcoming commands(request, httprequest).
                     result && result.request && (payload.context.request = result.request);
 
+                    visualizerResults = result && result.return && result.return.visualizer;
```
Comment (shamasis) [inline lib/runner/extensions/event.command.js]: extract visualiser results and bubble it via Item callback ... similar to how set next request decision is taken.

File: lib/runner/extensions/item.command.js [hunk 6]
```diff
@@ -8,7 +9,31 @@ var _ = require('lodash'),
...
+ * Returns visualizer data from the latest execution result.
+ *
+ * @param {Array} executions - Array of executions results
+ * @returns {*} - visualizer data
+ */
+extractVisualizerData = function (executions) {
+    var visualizerData,
+        i;
+
+    if (_.isArray(executions)) {
```
Comment (codenirvana) [inline lib/runner/extensions/item.command.js]: early bail out

File: lib/runner/extensions/item.command.js [hunk 7]
```diff
@@ -8,7 +9,31 @@ var _ = require('lodash'),
...
-    ALLOWED_REQUEST_MUTATIONS = ['url', 'method', 'headers'];
+    ALLOWED_REQUEST_MUTATIONS = ['url', 'method', 'headers'],
+
+    extractVisualizerData;
+
+/**
+ * Returns visualizer data from the latest execution result.
+ *
+ * @param {Array} executions - Array of executions results
+ * @returns {*} - visualizer data
```
Comment (codenirvana) [inline lib/runner/extensions/item.command.js]: Fix JSDoc

File: lib/visualizer/index.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
+var Handlebars = require('handlebars');
```
Comment (codenirvana) [inline lib/visualizer/index.js]: const

File: lib/runner/extensions/item.command.js [hunk 9]
```diff
@@ -152,8 +177,37 @@ module.exports = {
...
+                            // only extraxt visualizer data from results of pre-request scripts if
+                            // it is not found earlier
+                            if (!visualizerData) {
+                                visualizerData = extractVisualizerData(prereqExecutions);
+                            }
+
+                            if (visualizerData) {
+                                Visualizer.processTemplate(visualizerData.template,
+                                    visualizerData.data,
+                                    visualizerData.options,
```
Comment (codenirvana) [inline lib/runner/extensions/item.command.js]: don't expose options and the original template

File: lib/runner/extensions/item.command.js [hunk 10]
```diff
@@ -152,8 +177,37 @@ module.exports = {
...
+
+                                        // add processed template and data to visualizer results
+                                        visualizerData.processedTemplate = processedTemplate;
+                                    });
+                            }
+
                             // trigger an event saying that item has been processed
-                            this.triggers.item(null, coords, item); // @todo - should this trigger receive error?
+                            // @todo - should this trigger receive error?
+                            this.triggers.item(null, coords, item, visualizerData);
```
Comment (codenirvana) [inline lib/runner/extensions/item.command.js]:
```suggestion
                            this.triggers.item(null, coords, item, visualizer);
```

File: lib/runner/extensions/item.command.js [hunk 11]
```diff
@@ -152,8 +179,41 @@ module.exports = {
...
+                                        // add processed template and data to visualizer result
+                                        visualizerResult.processedTemplate = processedTemplate;
+                                        visualizerResult.data = visualizerData.data;
+                                    });
+                            }
+
                             // trigger an event saying that item has been processed
-                            this.triggers.item(null, coords, item); // @todo - should this trigger receive error?
+                            // @todo - should this trigger receive error?
+                            this.triggers.item(null, coords, item, visualizerResult);
```
Comment (codenirvana) [inline lib/runner/extensions/item.command.js]: It should be triggered within the `processTemplate` callback.

**Pull request merged**